python/ml4ir/applications/ranking/tests/test_t_test_calculation.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+from scipy import stats
+import unittest
+import warnings
+from ml4ir.base.model.relevance_model import perform_click_rank_dist_paired_t_test, compute_stats_from_stream
+
+warnings.filterwarnings("ignore")
+
+
+class TestTtestCalculation(unittest.TestCase):
+    """
+    Testing the calculation of paired t-test statistic and its p-value.
+    """
+
+    def t_test_calculation(self, a, b, batchzs, pvalue_threshold):
+        n = len(a)
+        expected_t_test_stat, expected_pvalue = stats.ttest_rel(a, b)
```

Comment:
this is a two-sided test. You are using a related sets test. Why are you not using a standard t-test? From the doc of scipy I see that it is like comparing the same set of students in two exams. The way I read it, a student is a ranker system and here you are comparing them on the same data. I admit, I do not expect that the p-val to be drastically different, I am just curious.

Reply:
The way I see it is the two models (or students) are evaluated on the same dataset with the same splits and so a paired t-test is a good fit.